compositor_pipeline/src/pipeline/output/mp4.rs

```diff
@@ -145,46 +149,56 @@ fn init_ffmpeg_output(
         })
         .transpose()?;
 
-    let audio_stream = options
-        .audio
-        .map(|a| {
-            let codec = ffmpeg::codec::Id::AAC;
-            let channels = match a.channels {
-                AudioChannels::Mono => 1,
-                AudioChannels::Stereo => 2,
-            };
-            let sample_rate = a.sample_rate as i32;
-
-            let mut stream = output_ctx
-                .add_stream(codec)
-                .map_err(OutputInitError::FfmpegMp4Error)?;
-
-            // If audio time base doesn't match sample rate, ffmpeg muxer produces incorrect timestamps.
-            stream.set_time_base(ffmpeg::Rational::new(1, sample_rate));
-
-            let codecpar = unsafe { &mut *(*stream.as_mut_ptr()).codecpar };
-            codecpar.codec_id = codec.into();
-            codecpar.codec_type = ffmpeg::ffi::AVMediaType::AVMEDIA_TYPE_AUDIO;
-            codecpar.sample_rate = sample_rate;
-            codecpar.ch_layout = ffmpeg::ffi::AVChannelLayout {
-                nb_channels: channels,
-                order: ffmpeg::ffi::AVChannelOrder::AV_CHANNEL_ORDER_UNSPEC,
-                // This value is ignored when order is AV_CHANNEL_ORDER_UNSPEC
-                u: ffmpeg::ffi::AVChannelLayout__bindgen_ty_1 { mask: 0 },
-                // Field doc: "For some private data of the user."
-                opaque: ptr::null_mut(),
-            };
+    let audio_stream = if let (Some(a), Some(encoder)) = (options.audio, encoder_config.audio) {
+        let codec = ffmpeg::codec::Id::AAC;
+        let channels = match a.channels {
+            AudioChannels::Mono => 1,
+            AudioChannels::Stereo => 2,
+        };
+        let sample_rate = a.sample_rate as i32;
+
+        let mut stream = output_ctx
+            .add_stream(codec)
+            .map_err(OutputInitError::FfmpegMp4Error)?;
+
+        // If audio time base doesn't match sample rate, ffmpeg muxer produces incorrect timestamps.
+        stream.set_time_base(ffmpeg::Rational::new(1, sample_rate));
+
+        let codecpar = unsafe { &mut *(*stream.as_mut_ptr()).codecpar };
+        let Mp4AudioEncoderConfig::Aac(encoder_config) = encoder;
+        unsafe {
+            codecpar.extradata = ffmpeg_next::ffi::av_malloc(encoder_config.len()) as *mut u8;
```

Comment:
@wkozyra95 Are you sure it should be `encoder_config.len() + AV_INPUT_BUFFER_PADDING_SIZE` instead of `encoder_config.len()`?

<img width="1897" alt="Screenshot 2025-03-06 at 14 05 32" src="https://github.com/user-attachments/assets/48cb89cb-5946-42a1-a1a1-9159fc613d5d" />

https://ffmpeg.org/doxygen/6.1/structAVCodecParameters.html#a9befe0b86412646017afb0051d144d13

https://github.com/membraneframework/membrane_rtmp_plugin/blob/50c3ad6cc43ebc42e3cdde3053c930b61bbfec42/c_src/membrane_rtmp_plugin/sink/rtmp_sink.c#L126